javalib/src/main/scala/java/time/Duration.scala

```diff
@@ -0,0 +1,232 @@
+package java.time
+
+import scala.collection.JavaConverters._
+
+import java.time.temporal._
+
+final class Duration private (seconds: Long, nanos: Int)
+    extends TemporalAmount with Comparable[Duration]
+    with java.io.Serializable with Preconditions {
+
+  require(nanos >= 0 && nanos <= 999999999, "nanos must be >= 0 and <= 999999999")
+
+  private val totalNanos: BigInt = BigInt(seconds) * 1000000000 + nanos
+
+  def get(unit: TemporalUnit) = unit match {
+    case ChronoUnit.SECONDS => seconds
+
+    case ChronoUnit.NANOS => nanos
+
+    case _ =>
```

Comment:
The specs say that this method could throw a `DateTimeException`, there might be something missing. I can't see that case where it would happen, but we should investigate this, it might be for a corner case.